docs/versioning.rst

```diff
@@ -0,0 +1,91 @@
+.. _versioning:
+
+Vyper Versioning Guideline
+##########################
+
+Motivation
+==========
+
+Vyper has different groups that are considered "users":
+
+- Smart Contract Developers (using Vyper to write smart contracts)
+- Package Integrators (integerating Vyper package or cli into their tools)
+- Security Professionals (aware of the status of various Vyper features and security issues)
+
+The sets of users need to understand what changes require their involvement or
+understanding of how the way they use Vyper will be impacted.
+This guideline will lay out how these different groups stay informed about the progress of Vyper.
+This guideline was adapted from `semantic versioning <https://semver.org/>`_.
+
+Smart Contract Developers
```

Comment:
This section, and the following two sections, could be handled as an intro paragraph and three bullet points (major, minor, patch) to make them less dense and easier to skim.